### PR TITLE
JWT all inherit from a base class

### PR DESCRIPTION
**Pull request by tmconnors**

### CODE REVIEW

File: lib/twilio-ruby/jwt/jwt.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
+module Twilio
+  module JWT
+    class BaseJWT
+      def initialize(secret_key: nil, issuer: nil, subject: nil, algorithm: 'HS256', nbf: nil, ttl: 3600, valid_until: nil)
+        @secret_key = secret_key
+        @issuer = issuer
+        @subject = subject
+        @algorithm = algorithm
```
Comment (kevinburke) [inline lib/twilio-ruby/jwt/jwt.rb]: I don't think it's a good idea for the algorithm to be configurable, anywhere. This is the root of pretty much every JWT vulnerability. There's also no reason to use any algorithm but HS256.

File: lib/twilio-ruby/jwt/jwt.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+module Twilio
+  module JWT
+    class BaseJWT
+      def initialize(secret_key: nil, issuer: nil, subject: nil, algorithm: 'HS256', nbf: nil, ttl: 3600, valid_until: nil)
```
Comment (jingming) [inline lib/twilio-ruby/jwt/jwt.rb]: We should probably document that we will favour `valid_until` over `ttl`.

File: lib/twilio-ruby/jwt/jwt.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        end
+
+        payload[:sub] = @subject unless @subject.nil?
+
+        return payload
+      end
+
+      def to_jwt
+        if @secret_key.nil?
+          raise ArgumentError, 'JWT does not have a signing key'
```
Comment (jingming) [inline lib/twilio-ruby/jwt/jwt.rb]: Is this a way to make this a required param in ruby?

File: lib/twilio-ruby/jwt/jwt.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        @issuer = issuer
+        @subject = subject
+        @algorithm = algorithm
+        @nbf = nbf
+        @ttl = ttl
+        @valid_until = valid_until
+      end
+
+      def _generate_headers
+        return {}
```
Comment (jingming) [inline lib/twilio-ruby/jwt/jwt.rb]: Ruby doesn't need `return` statements.

File: lib/twilio-ruby/jwt/jwt.rb (hunk 3 above)
Comment (ilanbiala) [inline lib/twilio-ruby/jwt/jwt.rb]: https://robots.thoughtbot.com/ruby-2-keyword-arguments#required-keyword-arguments

We would need to drop Ruby 2 (it's only supported in 2.1) if we want to keep it as a keyword argument.

Comment (tmconnors) [inline lib/twilio-ruby/jwt/jwt.rb]: It would still would require a not `nil` check, I can move this check to this initialization. @jingming @ilanbiala

Comment (ilanbiala) [inline lib/twilio-ruby/jwt/jwt.rb]: Guess not actually, still would need to check if it's `nil` or not..

**Pull request merged**